data_structures/stack_using_two_queues.cpp

```diff
@@ -0,0 +1,245 @@
+         * Pushes the given element to top of stack
+         */
+        void push(T data) { q1.push(data); }
+
+        /**
+         * Pops the topmost element of stack
+         */
+        void pop()
+        {
+            // push first n-1 elements from q1 to q2
+            // pop the last element from q1
+            // swap q1 and q2
+            if (q1.size() == 0)
+            {
+                return;
+            }
+
+            while (q1.size() > 1)
+            {
+                q2.push(q1.front());
+                q1.pop();
+            }
+            // pops the last element from q1;
+            q1.pop();
+
+            // swaps q1 with q2
+            q1.swap(q2);
+        }
+
+        /**
+         * Returns the topmost element of stack.
+         * Returns -1 if stack is empty
+         */
+        T top()
+        {
+            if (q1.size() == 0)
+            {
+                return T();
+            }
+
+            while (q1.size() > 1)
+            {
+                q2.push(q1.front());
+                q1.pop();
+            }
+
+            T top_element = q1.front();
+            q2.push(q1.front());
+            q1.pop();
+
+            // swaps q1 with q2
+            q1.swap(q2);
+
+            return top_element;
+        }
+
+        /**
+         * Returns the size of stack
+         */
+        int size() { return q1.size() + q2.size(); }
+
+        /**
+         * Returns true if stack is empty, otherwise false
+         */
+        bool empty() { return (q1.size() == 0); }
+    };
+
+} // namespace data_structures
+
+/**
+ * class Book with book name and it's price
+ */
+class Book
+{
+private:
+    std::string book_name{std::string()};
+    double book_price{double()};
+
+public:
+    /**
+     * default constructor
+     */
+    Book() = default;
+
+    /**
+     * parameterized constructor
+     */
+    Book(std::string book_name, double book_price)
+        : book_name{std::move(book_name)}, book_price{book_price}
+    {
+    }
+
+    /**
+     * Returns name of the book
+     */
+    std::string get_book_name() const
+    {
+        return this->book_name;
+    }
+
+    /**
+     * Returns price of the book
+     */
+    double get_book_price() const
+    {
+        return this->book_price;
+    }
+
+    /**
+     * comparison of two books
+     */
+    bool operator==(const Book &b) const
+    {
+        return ((b.get_book_name() == this->book_name) && (b.get_book_price() == this->book_price));
+    }
+};
+
+/**
+ * overloading operator<< to print book details
+ */
+std::ostream &operator<<(std::ostream &os, const Book &book)
+{
+    os << book.get_book_name() << ", " << book.get_book_price() << "\n";
+    return os;
+}
+
+/**
+ * utility function to print stack
+ */
+template <typename T>
+void print(data_structures::Stack<T> s)
+{
+    while (!s.empty())
+    {
+        std::cout << s.top() << " ";
+        s.pop();
+    }
+    std::cout << "\n";
+}
+
+/**
+ * Testing function
+ */
```

Comment:
```suggestion
 * @returns None
 */
```
This is just to ensure that the documentation gets generated.